crypto-ffi/bindings/js/CoreCrypto.ts

```diff
@@ -1567,18 +1567,17 @@ export class WireE2eIdentity {
      * Creates a new acme order for the handle (userId + display name) and the clientId.
      *
      * @param displayName human readable name displayed in the application e.g. `Smith, Alice M (QA)`
-     * @param domain DNS name of owning backend e.g. `example.com`
-     * @param clientId client identifier with user b64Url encoded & clientId hex encoded e.g. `impp:wireapp=NDUyMGUyMmY2YjA3NGU3NjkyZjE1NjJjZTAwMmQ2NTQ/6add501bacd1d90e@example.com`
-     * @param handle user handle e.g. `impp:wireapp=alice.smith.qa@example.com`
+     * @param clientId client identifier with user b64Url encoded & clientId hex encoded e.g. `NDUyMGUyMmY2YjA3NGU3NjkyZjE1NjJjZTAwMmQ2NTQ:6add501bacd1d90e@example.com`
+     * @param handle user handle e.g. `alice.smith.qa@example.com`
      * @param expiryDays generated x509 certificate expiry
      * @param directory you got from {@link directoryResponse}
      * @param account you got from {@link newAccountResponse}
      * @param previousNonce `replay-nonce` response header from `POST /acme/{provisioner-name}/new-account`
      * @see https://www.rfc-editor.org/rfc/rfc8555.html#section-7.4
      */
-    newOrderRequest(displayName: string, domain: string, clientId: string, handle: string, expiryDays: number, directory: AcmeDirectory, account: AcmeAccount, previousNonce: string): JsonRawData {
+    newOrderRequest(displayName: string, clientId: ClientId, handle: string, expiryDays: number, directory: AcmeDirectory, account: AcmeAccount, previousNonce: string): JsonRawData {
```

Comment:
If the E2EI client ID is a string, wouldn't it be better to use a `string` in user-facing APIs and then do the conversion ourselves at the FFI boundary?

Reply:
On one hand this is true for e2e identity since it has to know the clientId format to parse it.  
On the other hand, MLS part has to ignore the clientId format. Requiring a String could look like an unrequired constraint.

The goal of this was to have a uniform API from a consumer POV. So we either choose `Vec<u8>` or `String`.